File: hackerrank/oct13/angry-children.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
  /* Enter your code here. Read input from STDIN. Print output to STDOUT */
  int N,K;
  cin>>N>>K;
  vector <int> A(N);
  for(int i=0;i<N;i++) cin>>A[i];
  sort(A.begin(),A.end());
  int min=A[K-1]-A[0];
  for(int i=0;i<N-K;i++) {
    int curr=A[i+K-1]-A[i];
    if(min>curr) min=curr;

  }
  cout<<min<<endl;
  return 0;
}

File: hackerrank/challenges/angry-children.py
N = int(raw_input())
K = int(raw_input())
A = [int(raw_input()) for _ in xrange(N)]
A.sort()
#print A
m = A[-1] - A[0]
for i in xrange(N-K+1):
    m = min(m,A[i+K-1]-A[i])
print m
